protocol/messenger_handler.go

```diff
@@ -406,20 +412,114 @@ func (m *Messenger) handleCommandMessage(state *ReceivedMessageState, message *c
+	}
+
 	for _, contact := range message.Contacts {
-		err := m.HandleSyncInstallationContact(state, *contact)
+		err = m.HandleSyncInstallationContact(state, *contact)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
 
 	for _, community := range message.Communities {
-		err := m.handleSyncCommunity(state, *community)
+		err = m.handleSyncCommunity(state, *community)
 		if err != nil {
-			return err
+			errors = append(errors, err)
 		}
 	}
+	err = m.handleSyncSettings(state, message.Settings)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	if m.config.messengerSignalsHandler != nil {
+		response := wakusync.SyncFromWakuResponse{}
+
+		response.AddWakuSyncDetails(SyncWakuSectionKeyProfile, message.ProfileDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeyContacts, message.ContactsDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeyCommunities, message.CommunitiesDetails)
+		response.AddWakuSyncDetails(SyncWakuSectionKeySettings, message.SettingsDetails)
+
+		m.config.messengerSignalsHandler.SendSyncFromWakuProgress(&response)
+	}
+
+	return errors
+}
+
+func (m *Messenger) handleSyncProfile(state *ReceivedMessageState, message *protobuf.SyncProfile, backupTime uint64) error {
+	if message == nil {
+		return nil
+	}
+
+	pubKey := common.PubkeyToHex(&m.identity.PublicKey)
+	contact, ok := state.AllContacts.Load(pubKey)
+	if !ok {
+		return errors.New("cannot find logged in profile among all known contacts")
+	}
+
+	response := wakusync.SyncFromWakuResponse{
+		Profile: &wakusync.SyncProfile{},
+	}
+
+	if contact.LastUpdated < backupTime {
+		err := m.SetDisplayName(message.DisplayName, false)
+		response.AddDisplayName(message.DisplayName, err == nil)
+	}
+
+	syncWithBackedUpImages := false
+	dbImages, err := m.multiAccounts.GetIdentityImages(message.KeyUid)
+	if err != nil {
+		syncWithBackedUpImages = true
+	}
+	if len(dbImages) == 0 {
+		if len(message.Pictures) > 0 {
+			syncWithBackedUpImages = true
+		}
+	} else {
+		// since both images (large and thumbnail) are always stored in the same time, we're free to use either of those two clocks for comparison
+		lastImageStoredClock := dbImages[0].Clock
+		syncWithBackedUpImages = lastImageStoredClock < backupTime
+	}
+
+	if syncWithBackedUpImages {
+		if len(message.Pictures) == 0 {
+			err = m.multiAccounts.DeleteIdentityImage(message.KeyUid)
+			response.AddImages(nil, err == nil)
+		} else {
+			idImages := make([]images.IdentityImage, len(message.Pictures))
+			for i, pic := range message.Pictures {
+				img := images.IdentityImage{
+					Name:         pic.Name,
+					Payload:      pic.Payload,
+					Width:        int(pic.Width),
+					Height:       int(pic.Height),
+					FileSize:     int(pic.FileSize),
+					ResizeTarget: int(pic.ResizeTarget),
+					Clock:        pic.Clock,
+				}
+				idImages[i] = img
+			}
+			err = m.multiAccounts.StoreIdentityImages(message.KeyUid, idImages, false)
+			response.AddImages(idImages, err == nil)
```

Comment:
Similarly here, if there's any error but we want to continue, probably best to leave comments explaining why and what we are trying to achieve

Reply:
explained above